arithmetic-engine/src/main/kotlin/com/raccoon/backend/arithmetic/internal/SimpleParser.kt

```diff
@@ -0,0 +1,133 @@
+package com.raccoon.backend.arithmetic.internal
+
+import com.raccoon.backend.arithmetic.UnexpectedTokenException
+
+/**
+ * A simple arithmetic parser.
+ * Parses a given string and evaluates the expression it contains.
+ *
+ *  Grammar:
+ *  <Input>   : <Expr>EOF
+ *  <Expr>    : <Term><Terms>
+ *  <Terms>   : +<Term><Terms> | -<Term><Terms> | empty
+ *  <Term>    : <Factor><Factors>
+ *  <Factors> : *<Factor><Factors> | /<Factor><Factors> | empty
+ *  <Factor>  : constant | (<Expr>) | -<Factor> | +<Factor>
+ */
+class SimpleParser(
+    private val lexerFactory: LexerFactory = SimpleLexer.factory,
+    private val stackMachine: StackMachine = SimpleStackMachine()
+) {
+    private lateinit var lexer: Lexer
+    private lateinit var input: String
+
+    private fun error(): Nothing = throw UnexpectedTokenException(input, lexer.lastToken.offset)
+    private fun next() = lexer.next()
+
+    private fun init(input: String) {
+        stackMachine.reset()
+        this.input = input
+        lexer = lexerFactory.create(input)
+        lexer.next() // Read the first token.
+    }
+
+    /**
+     * Parses the [input] string and evaluate the expression it contains.
+     * Resets the stack machine before parsing.
+     * @return the result of the evaluation.
+     * @throws com.raccoon.backend.arithmetic.ParsingException - if the input expression is incorrect.
+     */
+    fun parse(input: String): Double {
+        init(input)
+        expr()
+        eof()
+        return stackMachine.dPop()
+    }
+
+    private fun eof() {
```

Comment:
I think renaming this method is included in the previous comment (e.g. `eof` -> `processEof`/`parseEof`). Don't you mind if I mark this conversation as resolved since this point is already tracked by another comment?